aws/aws_test.go

```diff
@@ -175,6 +177,92 @@ func TestEnsureQueuePolicy(t *testing.T) {
 	}
 }
 
+func TestChangeMessageRetentionPeriod(t *testing.T) {
+	queueURL := aws.String("queueURL")
+	testCases := []struct {
+		Duration                time.Duration
+		CallsAWS                bool
+		ExpectedErr             error
+		ExpectedRetentionPeriod int64
+	}{
+		{Duration: 5 * time.Minute, CallsAWS: true, ExpectedRetentionPeriod: 300},
+		{Duration: -1 * time.Second, ExpectedErr: ErrMessageRetentionPeriodOutOfRange},
+		{Duration: 1000 * time.Hour, ExpectedErr: ErrMessageRetentionPeriodOutOfRange},
+		{Duration: 60 * time.Second, ExpectedErr: errors.New("expected SetQueueAttributes error"), CallsAWS: true},
```

Comment:
it would be useful to add test cases that exercise the boundary conditions -- in this case that would be 60 and 1209600